kusd/handler.go

```diff
@@ -628,48 +628,36 @@ func (pm *ProtocolManager) handleMsg(p *peer) error {
 		pm.txpool.AddRemotes(txs)
 
 	case msg.Code == ProposalMsg:
-		// @TODO (rgeraldes) - review flow (we will not need this condition)
-		if !pm.validator.Validating() {
-			break
-		}
-
 		// Retrieve and decode the propagated proposal
 		var proposal types.Proposal
 		if err := msg.Decode(&proposal); err != nil {
 			return errResp(ErrDecode, "msg %v: %v", msg, err)
 		}
-		pm.validator.AddProposal(&proposal)
-
-	case msg.Code == VoteMsg:
-		// @TODO (rgeraldes) - review flow (we will not need this condition)
-		if !pm.validator.Validating() {
+		if err := pm.validator.AddProposal(&proposal); err != nil {
```

Comment:
Should we add some log if this returns an `err`?

Reply:
I think so, either log or comment saying why we ignoring the error

Reply:
on the same file pm.txpool.AddRemotes(txs) throws an error but they are not handling the error. Have a look at their flow to see if we can do something similar.

Reply:
not sure if I follow, they are ignoring the errors, what should we do then?